### PR TITLE
Cache progress and responses.

### PR DESCRIPTION
**Pull request by charleswaddell**
On the account details page for the RAP sites we do query nearly every
progress and response object so let's start by caching them right away.

PT Story: https://www.pivotaltracker.com/story/show/98337362

### CODE REVIEW

File: CME/dataobjects/CMEAccount.php [hunk 1]
```diff
@@ -71,11 +92,16 @@ public function isQuizComplete(CMECredit $credit)
 		$progress = $this->getCMEProgress($credit);
 
 		if ($progress instanceof CMEAccountCMEProgress &&
-			$progress->quiz instanceof CMEQuiz) {
+			$progress->hasInternalValue('quiz')) {;
```
Comment (nburka) [inline CME/dataobjects/CMEAccount.php]: semi-colon?

**Comment by charleswaddell** [comment]
Goes with silverorange/Rap#792

File: CME/dataobjects/CMEAccount.php [hunk 2]
```diff
@@ -90,8 +116,19 @@ public function isQuizPassed(CMECredit $credit)
 
 		if ($this->isQuizComplete($credit)) {
 			$progress = $this->getCMEProgress($credit);
-			$quiz_response = $progress->quiz->getResponseByAccount($this);
-			$passed = $quiz_response->isPassed();
+
+			if ($progress instanceof CMEAccountCMEProgress &&
+				$progress->hasInternalValue('quiz')) {;
```
Comment (nburka) [inline CME/dataobjects/CMEAccount.php]: semi-colon?

File: CME/dataobjects/CMEAccount.php [hunk 3]
```diff
@@ -49,9 +67,12 @@ public function isEvaluationComplete(CMECredit $credit)
 		$progress = $this->getCMEProgress($credit);
 
 		if ($progress instanceof CMEAccountCMEProgress &&
-			$progress->evaluation instanceof CMEEvaluation) {
+			$progress->hasInternalValue('evaluation')) {;
```
Comment (nburka) [inline CME/dataobjects/CMEAccount.php]: semi-colon?

File: CME/dataobjects/CMEAccount.php [hunk 4]
```diff
@@ -244,23 +281,99 @@ public function getCMEProgress(RapCredit $credit)
...
+			$this->response_by_cme_eval[] = array();
+
+			$sql = sprintf(
+				'select * from InquisitionResponse
+				where account = %s and reset_date is null',
+				$this->db->quote($this->id, 'integer')
+			);
+
+			$responses = SwatDB::query($this->db, $sql,
+				SwatDBClassMap::get('CMEEvaluationResponseWrapper'));
```
Comment (nburka) [inline CME/dataobjects/CMEAccount.php]: cs

File: CME/dataobjects/CMEAccount.php [hunk 5]
```diff
@@ -244,23 +281,99 @@ public function getCMEProgress(RapCredit $credit)
...
+			$this->response_by_cme_quiz[] = array();
+
+			$sql = sprintf(
+				'select * from InquisitionResponse
+				where account = %s and reset_date is null',
+				$this->db->quote($this->id, 'integer')
+			);
+
+			$responses = SwatDB::query($this->db, $sql,
+				SwatDBClassMap::get('CMEQuizResponseWrapper'));
```
Comment (nburka) [inline CME/dataobjects/CMEAccount.php]: cs

File: CME/dataobjects/CMEQuizResponse.php [hunk 6]
```diff
@@ -69,6 +63,27 @@ public function isPassed()
 
 	public function getCredits()
 	{
+		return $this->credits;
```
Comment (nburka) [inline CME/dataobjects/CMEQuizResponse.php]: can we drop this method altogether and force the use of `loadCredits()`?

**Pull request merged**